Fix Server getlegendgraphic duplicated symbols

elpaso opened the pull request:
Fix Server getlegendgraphic duplicated symbols
... and fix a few build warnings

Now uses the hit test from QGIS API.

Also: WMS contextual legend: fix unreported issue with reprojection:

When using contextual WMS legends with a different projection
in the canvas and in the layer, the extent was not transformed.

elpaso committed:
WMS contextual legend: fix unreported issue with reprojection

When using contextual WMS legends with a different projection
in the canvas and in the layer, the extent was not transformed.

elpaso committed:
Fix Server getlegendgraphic duplicated symbols

... and fix a few build warnings

Now uses the hit test from QGIS API.

Fixes #32020

elpaso committed:
Code layout and mask image 1

elpaso committed:
Control image mask 2

elpaso committed:
Added control image 3

nyalldawson commented on src/providers/wms/qgswmsprovider.cpp:
```diff
@@ -3461,6 +3461,8 @@ QgsImageFetcher *QgsWmsProvider::getLegendGraphicFetcher( const QgsMapSettings *
     scale = mapSettings->scale();
     mapExtent = mapSettings->visibleExtent();
+    QgsCoordinateTransform ct { mapSettings->destinationCrs(), crs(), mapSettings->transformContext() };
+    mapExtent = ct.transformBoundingBox( mapExtent );
```
Better catch QgsCsException here

elpaso committed:
Catch cs exception and add another control file mask

elpaso committed:
Add another control image mask

elpaso committed:
Oouch (again) on that silly WMS 1.3.0 inverted axis

A reminder for me (from geoserver docs)

The WMS 1.3.0 specification mandates that the axis ordering for geographic coordinate systems defined in the EPSG database be latitude/longitude, or y/x. This is contrary to the fact that most spatial data is usually in longitude/latitude, or x/y. This requires that the coordinate order in the BBOX parameter be reversed for SRS values which are geographic coordinate systems.
